proto/buf/alpha/registry/v1alpha1/doc.proto

```diff
@@ -377,6 +378,47 @@ message FieldOptions {
   int32 jstype = 4;
 }
 
+// FieldLiteral represents a field and its value. It can be used to model descriptor options.
+message FieldLiteral {
+  // When is_extension is true, name is the fully-qualified name of the
+  // extension. Otherwise, it is the simple name of the field.
+  string name = 1;
+  // The unique field number associated with the field.
+  int32 tag = 2;
+  // Indicates whether this is an extension or not.
+  bool is_extension = 3;
+  // The field's value.
+  LiteralValue value = 4;
+}
+
+// MessageLiteral represents an option value for a type that is a message.
+message MessageLiteral {
+  // Map of field names to value. Extension field names will be in the form:
+  // [fully.qualified.extension.Name].
+  repeated FieldLiteral fields = 1;
+}
+
+// ArrayLiteral represents an option value for a type that is a repeated field.
+// Maps are also represented using an ArrayLiteral, where each map entry is a MessageLiteral
+// that contains a key and value field.
+message ArrayLiteral {
+  repeated LiteralValue elements = 1;
+}
+
+// LiteralValue represents the actual value of the extension.
+message LiteralValue {
+  oneof value {
+    string string_value = 2;
+    int64 int_value = 3;
```

Comment:
Why does this not contain all the primitive types? This API shouldn't be frontend-specific.

Reply:
updated this to include all the primitive types

Reply:
@bufdev, is this API really not frontend specific? The comment on the service says otherwise: https://github.com/bufbuild/buf/blob/17b4060d969e99b151d125ff0d5ff007aa271250/proto/buf/alpha/registry/v1alpha1/doc.proto#L19

Either way, the API is undoubtedly specific to docs, and we ultimately want to present the option value as text. I cannot think of a use case where I would want to distinguish even `DOUBLE` from `FLOAT` in the context of docs, and I'd prefer if we use the widest types only. I don't think it's an issue to include all types here, just that I don't see the need.

Reply:
In general, APIs like this (and APIs in general) really shouldn't be designed with a specific caller in mind, but in this specific instance I don't know how much the field difference matters.